multiversx_sdk_core/transaction_builders/delegation_factory.py

```diff
@@ -0,0 +1,430 @@
+
+        if unset:
+            data.append(arg_to_string('false'))
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=data,
+            execution_gas_limit=MetaChainSystemSCsCost.DELEGATION_OPS + ADDITIONAL_GAS_FOR_OPERATIONS,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def redelegate_cap(self,
+                       sender: IAddress,
+                       delegation_contract: IAddress,
+                       set: bool,
+                       unset: bool,
+                       value: ITransactionValue,
+                       transaction_nonce: Optional[INonce] = None,
+                       gas_price: Optional[IGasPrice] = None,
+                       gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        data = ["setCheckCapOnReDelegateRewards"]
+
+        if set:
+            data.append(arg_to_string('true'))
+
+        if unset:
+            data.append(arg_to_string('false'))
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=data,
+            execution_gas_limit=MetaChainSystemSCsCost.DELEGATION_OPS + ADDITIONAL_GAS_FOR_OPERATIONS,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def set_metadata(self,
+                     sender: IAddress,
+                     delegation_contract: IAddress,
+                     name: str,
+                     website: str,
+                     identifier: str,
+                     value: ITransactionValue,
+                     transaction_nonce: Optional[INonce] = None,
+                     gas_price: Optional[IGasPrice] = None,
+                     gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        data = [
+            "setMetaData",
+            arg_to_string(name),
+            arg_to_string(website),
+            arg_to_string(identifier)
+        ]
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=data,
+            execution_gas_limit=MetaChainSystemSCsCost.DELEGATION_OPS + ADDITIONAL_GAS_FOR_OPERATIONS,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def _compute_gas_limit(self, payload: ITransactionPayload, execution_gas: IGasLimit) -> IGasLimit:
+        data_movement_gas = self.config.min_gas_limit + self.config.gas_limit_per_byte * payload.length()
+        return data_movement_gas + execution_gas
+
+    def _parse_keys(self, keys: List[str]) -> Tuple[str, int]:
```

Comment:
Is the function still used?

Reply:
Not used anymore. Forgot to delete it. It's deleted now.